### ISSUE TITLE
`expect_lint_free` warns but doesn't lint

### ISSUE THREAD

**Issue by ezraporter** [r-lib/lintr#1913]
In version `3.0.2` `lintr::expect_lint_free()` throws a warning but doesn't lint the package.

`tests/testthat/test-lint-free.R`:
```r
test_that("package is lint free", {
  lintr::expect_lint_free()
})
```

Console:
```r
devtools::test()

#> ✔ | F W S  OK | Context 
#> ✔ |   1     1 | lint-free                                                                                                                                   
#> ────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
#> Warning (test-lint-free.R:2): package is lint free
#> Didn't find any R package searching upwards from '.'.
#> Backtrace:
#>  1. lintr::expect_lint_free()
#>       at test-lint-free.R:2:2
#>  2. lintr::lint_package(...)
```

**Comment by ezraporter** [comment]
Apologies. This is fixed in the development version. I should have checked before opening.
